src/python/m5/ext/pystats/jsonloader.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2020 The Regents of The University of California
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met: redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer;
+# redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution;
+# neither the name of the copyright holders nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .simstat import SimStat
+from .statistic import Scalar, Distribution, Accumulator
+from .group import Group, Vector
+import json
+
+class JSONLoader(json.JSONDecoder):
+    """
+    Subclass of JSONDecoder that overrides 'object_hook'. Converts JSON object
+    into a SimStat object.
+
+    Usage
+    -----
+    ```
+    from m5.ext.pystats.jsonloader import JSONLoader
+
+    with open(path) as f:
+    simstat_object = json.load(f, cls=JSONLoader)
+    ```
+    """
+
+    def __init__(self):
+        json.JSONDecoder.__init__(self, object_hook=self.__json_to_simstat)
+
+    def __json_to_simstat(self, d: dict) -> SimStat:
+        if 'type' in d:
+            if d['type'] == 'Scalar':
+                d.pop('type', None)
```

Comment:
I suggest dropping these. Instead, I think that `Scalar`, `Vector`, etc. should be modified something like below...

```
class Scalar(Statistic):
    """
    A scalar Python statistic type.
    """

    value: Union[float, int]

    def __init__(self, value: Any,
                 unit: Optional[str] = None,
                 description: Optional[str] = None,
                 datatype: Optional[StorageType] = None,
                 **kwargs = None):
        super(Scalar, self).__init__(
                                     value=value,
                                     type="Scalar",
                                     unit=unit,
                                     description=description,
                                     datatype=datatype,
                                     **kwargs
                                    )
```

Reply:
Does this solve the problem? (I will try this in just a second to find out.) I don't think the issue was that the constructor wouldn't take additional arguments, but it was because it was receiving two of the same argument, e.g. 'type' was received twice?

Reply:
I *think* so. I meant to tag @BobbyRBruce to see what he thinks about this as well. 

I could be wrong, though

Reply:
I tried it, and ran into the same error from before I added the d.pop() statements:

`  File "build/NULL/python/m5/ext/pystats/jsonloader.py", line 53, in __json_to_simstat
  File "build/NULL/python/m5/ext/pystats/statistic.py", line 72, in __init__
TypeError: __init__() got multiple values for keyword argument 'type'`

Reply:
Gotcha... Oh! I see... 
```
( value=value, type="Scalar",
...
```

I'm good with this change as it is, then :)